### ISSUE TITLE
2D-CNN or 3D-CNN?

### ISSUE THREAD

**Issue by atonyo11** [hulianyuyy/CorrNet#15]
I apologize if I misunderstood. As shown in the picture, the feature extractor is depicted as a 2D CNN. However, in the code, a ResNet-based model is built using 3D CNN components, such as conv3x3, BatchNorm3d, etc. Could you please explain why the feature extractors are implemented using 2D CNN? Thank you in advance.

![image](https://github.com/hulianyuyy/CorrNet/assets/149935186/5bb16f53-f3e3-4eca-b4bd-6f7396a1945b)

**Comment by hulianyuyy** [comment]
We always set the temporal kernel size as 1, and thus it's still a 2D CNN.



---Original---
From: ***@***.***&gt;
Date: Mon, Nov 13, 2023 21:14 PM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;
Subject: [hulianyuyy/CorrNet] 2D-CNN or 3D-CNN? (Issue #15)




 
Sorry if I miss understand. As show in the picture, feature extractor is 2D CNN, but in the code, resnet based is build from 3D CNN, eg. conv3x3, BatchNorm3d etc. So why it is feature extractor is 2D CNN? Thank you in advance.

**Comment by atonyo11** [comment]
> temporal kernel size as 1
As I understand, when kernel_size=(1,7,7) in the line `self.conv1 = nn.Conv3d(3, 64, kernel_size=(1,7,7), stride=(1,2,2), padding=(0,3,3), bias=False)`, it functions as a 2D CNN. Is it right?

**Comment by atonyo11** [comment]
I got it. Thank you very much!
